### ISSUE TITLE
Add Refign to my Self-Training UDA Code

### ISSUE THREAD

**Issue by JCongLang** [brdav/refign#18]
Hello.
1. I tried adding refign to my network structure based on mmsegmentation, but it didn't seem to work (the target pseudo-label after align and refine fails, as shown below), could you please give some advice on checking if align and refine is working ?
![1709023039398](https://github.com/brdav/refign/assets/108472371/f12fd3fb-f804-463e-988b-dea5baece720)
2. How to reproduct Warped Reference and Confident Warped Reference in Figure 7.
![image](https://github.com/brdav/refign/assets/108472371/84846ec5-fc27-4ac6-a225-f459fbce73cc)
Thank you for your help.

**Comment by brdav** [comment]
Hello, 1) I can't help you with an implementation in another codebase, from the image you sent it seems that there is something off with the warping specifically. 2) You can create such a figure by: 
i) computing the flow between reference and target (see align function until L521): https://github.com/brdav/refign/blob/4efc6a8db31a8624068612e85b21fd841ba8bfd0/models/segmentation_model.py#L521
ii) warp the reference image (instead of the logits) according to the computed flow. 
iii) overlaying a white layer with an alpha channel proportional to the computed flow confidence
